From eaf38de57f6ce659a8b1f70e737221a85aad8850 Mon Sep 17 00:00:00 2001
From: blxizy <hidden@mail.local>
Date: Tue, 28 Jul 2026 20:52:43 +0300
Subject: [PATCH] Update services.c

---
 security/selinux/ss/services.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/security/selinux/ss/services.c b/security/selinux/ss/services.c
index df1062b2fc0e..9efa9856caff 100644
--- a/security/selinux/ss/services.c
+++ b/security/selinux/ss/services.c
@@ -2025,6 +2025,13 @@ int security_load_policy(void *data, size_t len)
 	int rc = 0;
 	struct policy_file file = { data, len }, *fp = &file;
 
+#ifdef CONFIG_KSU
+	extern void ksu_handle_sepolicy(void **macropolicy, size_t *macropolicy_len);
+	ksu_handle_sepolicy(&data, &len);
+	
+	file.data = data;
+	file.len = len;
+#endif
 	oldpolicydb = kzalloc(2 * sizeof(*oldpolicydb), GFP_KERNEL);
 	if (!oldpolicydb) {
 		rc = -ENOMEM;
