diff --git a/kernel/feature/adb_root.c b/kernel/feature/adb_root.c
index 0f71a50d..ddb52d1c 100644
--- a/kernel/feature/adb_root.c
+++ b/kernel/feature/adb_root.c
@@ -16,6 +16,10 @@
 
 #include "klog.h" // IWYU pragma: keep
 
+#ifndef ALIGN_DOWN
+#define ALIGN_DOWN(x, a) ALIGN((x) - ((a) - 1), (a))
+#endif
+
 DEFINE_STATIC_KEY_FALSE(ksu_adb_root);
 
 static long is_exec_adbd(struct pt_regs *regs)
